### PR TITLE
vacation2019/ts

### LINKED ISSUE #11065: Migrate `parts-more/BoxPlotSeries` to TypeScript

### LINKED ISSUE #11067: Migrate `parts-more/BubbleSeries` to TypeScript

### LINKED ISSUE #11071: Migrate `parts-more/ErrorBarSeries` to TypeScript

### LINKED ISSUE #11076: Migrate `parts-more/PolygonSeries` to TypeScript

### LINKED ISSUE #11078: Migrate `parts-more/WaterfallSeries` to TypeScript

### PR DESCRIPTION
**Pull request by bre1470**
- Fixed #11067 - Migrated BubbleSeries to TypeScript.
- Fixed #11065 - Migrated BoxPlotSeries to TypeScript.
- Fixed types of data labels options.
- Fixed #11071 - Migrated ErrorBarSeries to TypeScript.
- Fixed #11076 - Migrated PolygonSeries to TypeScript.
- Fixed #11078 - Migrated WaterfallSeries to TypeScript.

### CODE REVIEW

File: ts/parts/PieSeries.ts [hunk 1]
```diff
@@ -821,15 +821,17 @@ seriesType<Highcharts.PieSeriesOptions>(
...
+            this: Highcharts.PieSeries,
             y: number,
             left: boolean,
             point: Highcharts.PiePoint
         ): number {
             var center = this.center,
                 // Variable pie has individual radius
-                radius = this.radii ?
-                    this.radii[point.index as any] :
+                radius = (this as Highcharts.VariablePieSeries).radii ?
```
Comment (pawelfus) [inline ts/parts/PieSeries.ts]: Shouldn't this be `PieSeries` too? `PieSeries` is base series, `VariablePieSeries` is created on top of `PieSeries`

Comment (bre1470) [reply ts/parts/PieSeries.ts]: PieSeries has no radii property.

Comment (bre1470) [reply ts/parts/PieSeries.ts]: This happens, if you do no proper class inheritance with function masking and instead modify the base function.

Comment (pawelfus) [reply ts/parts/PieSeries.ts]: Ah, I see now. Thanks for the explanation!

**Pull request merged**